app/models/user.rb

```diff
@@ -1,14 +1,17 @@
 class User < ApplicationRecord
-  devise :database_authenticatable, :rememberable
+  devise :database_authenticatable
   belongs_to :territory
-  has_many :contacts
+  has_many :contacts, dependent: :destroy
 
-  validates_presence_of :email, :name, :password
+  validates_presence_of :email, :name, :password, on: :create
   validates :email, :format => Devise.email_regexp
-  validates :role, inclusion: { in: %w(admin, staff), message: "%{value} is not a valid role"}
-
+  validates :role, inclusion: { in: %w(admin staff), message: "%{value} is not a valid role"}
 
   def is_admin?
     self.role == 'admin'
   end
+
+  def get_order_target
+    self.order_target
```

Comment:
No use of this. I added this for some purpose initially but since there is no requirement of this, we will have to remove it.